drivers/neural_net/intel_gna.c

```diff
@@ -111,23 +111,27 @@ static void intel_gna_interrupt_handler(struct device *dev)
 
 		k_msgq_put(&gna->response_queue, &pending_resp, K_NO_WAIT);
 
-		k_alert_send(&gna->gna_cb_alert);
+		k_work_submit(&gna->gna_work);
 	}
 
 	/* clear GNA operation and disable interrupt */
 	regs->gnactrl |= GNA_CTRL_INTR_DISABLE | GNA_CTRL_ABORT_CLEAR;
 	gna->state = GNA_STATE_IDLE;
 }
 
-static int gna_cb_alert_handler(struct k_alert *alert)
+static void gna_work_handler(struct k_work *work)
 {
-	struct intel_gna_data *gna = (struct intel_gna_data *)alert;
-	struct intel_gna_pending_resp response;
+	struct intel_gna_data *gna = (struct intel_gna_data *)work;
+	int rc;
 
-	k_msgq_get(&gna->response_queue, &response, K_NO_WAIT);
+	do {
+		struct intel_gna_pending_resp response;
 
-	response.callback(&response.response);
-	return 0;
+		rc = k_msgq_get(&gna->response_queue, &response, K_NO_WAIT);
+		if (rc == 0) {
+			response.callback(&response.response);
+		}
+	} while (rc == 0);
```

Comment:
Does the system workqueue call the handler once for every `k_work_submit`?
If yes, then the do-while loop is not necessary.

Reply:
No, it calls it once.  The queue doesn't capture how much work is to be done, so the loop is required.

Reply:
Revised.  Thanks for the review.

Reply:
Thanks for the change. I'm getting a compile error (undeclared variable.
Please rename `response.callback()` to `resp.callback()`.

Reply:
Sorry about that; I had to shorten the variable name to conform to Zephyr's line width requirements without wrapping the condition expression and wasn't able to find a test case to build to check it.  Should be fixed now.

Reply:
I understand. It builds now.